interview-questions.md

```diff
@@ -7,90 +7,112 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. Why would you use the method `super()`?
 
-  Your answer:
+  Your answer: 
+    super() is used if you're creating a new class that you want to inherit an existing class's constructors.
 
-  Researched answer:
+  Researched answer: 
+    The method super() is used when extending a parent class into a new child class you want to create. Using super() in the child class will inherit the parent class' constructors. super() can also be used to call on methods in parent classes within child classes.
 
 
 
 2. What is the DOM? What is the virtual DOM?
 
-  Your answer:
+  Your answer: 
+    DOM is Document Object Model. Not sure what virtual DOM specifically refers to.
 
   Researched answer:
+    The DOM is an object-oriented representation of your page, that can be modified with a scripting language like JavaScript. A virtual DOM refers to an "ideal" version of your page. This is synced with the actual DOM for your page with something like ReactDOM. This might be wrong, still a little shakey on DOM.
 
 
 
 3. What is a higher-order function?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: 
+    The only two higher-order functions that come to mind are .map() and .filter which return new arrays after manipulating the data. 
```

Comment:
Great! But in the given scenario of having a conversation with an interviewer, the person is asking not specifically for higher-order function names, but what a higher-order function IS.